### ISSUE TITLE
Bad code generated for vpslld,vpsrld,vpsrad etc.

### ISSUE THREAD

**Issue by bpxhmemcpy** [Terraspace/HJWasm#180]
v2.55

```asm
vpslld ymm15, ymm0, 0  ;; vpsrld,vpsrad etc. same issue
``` 
generates:
```asm
C4 C1 05 72 F0 00    vpslld      ymm15,ymm8,0
```
should be:
```asm
db 0xC5, 0x85, 0x72, 0xF0, 0x0
```

**Comment by john-terraspace** [comment]
Fixed in 2.56, a promotion to 3byte vex prefix wasn't considering instr. forms with 2 registers + imm in the legacy codegen.

**Comment by bpxhmemcpy** [comment]
Thanks! I tested 2.56 and can confirm it's fixed.
